
Here there are pipelines for how reads were aligned.

We used ATAC-seq fastqs PRJNA667175 and RNA-seq fastqs liver (PRJNA591630) and testis (PRJNA591630) from threespine stickleback fish.

The starting input are fastqs, that can be processed and aligned into a .bam alignment of ATAC and transcriptomes. The .bam ATAC file is used to generate peaks.

For more detailed scripts related to ATAC-seq alignment see https://github.com/daniel-shaw1/ATAC-seq

